nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/autodiff/opvalidation/LayerOpValidation.java

```diff
@@ -1466,5 +1471,183 @@ public void testBiasAdd_nchw_nhwc() {
             String err = OpValidation.validate(tc);
             assertNull(err);
         }
+    }
+
+
+    @Test
+    public void LSTMLayerTestCase1() {
+
+        int miniBatchSize = 10;
+        int numUnits = 10;
+        int maxTSLength = 64;
+        int sL = 10; //small just for test
+
+        SameDiff sd = SameDiff.create();
+
+        // notations:
+        // bS - batch size
+        // sL - sequence length, number of time steps
+        // nIn - input size
+        // nOut - output size (hidden size)
+
+
+        // [bS, sL, nIn]  when dataFormat == 2 (NST)
+
+        SDVariable in = sd.var("in", Nd4j.rand( miniBatchSize, sL, numUnits));
+
+
+        SDVariable cLast = sd.var("cLast", Nd4j.zeros(DataType.FLOAT, miniBatchSize, numUnits));
+        SDVariable yLast = sd.var("yLast", Nd4j.zeros(DataType.FLOAT, miniBatchSize, numUnits));
+
+
+        List<SDVariable> outputs = sd.rnn.lstmLayer(
+                maxTSLength, in, cLast, yLast,
+                LSTMLayerWeights.builder()
+                        .weights(sd.var("weights", Nd4j.rand(numUnits, 4 * numUnits)))
+                        .rWeights(sd.var("rWeights", Nd4j.rand(numUnits, 4 * numUnits)))
+                        .peepholeWeights(sd.var("inputPeepholeWeights", Nd4j.rand(DataType.FLOAT, 3 * numUnits)))
+                        .bias(sd.var("bias", Nd4j.rand(DataType.FLOAT, 4 * numUnits))).build(),
+                LSTMLayerConfig.builder()
+                        .lstmdataformat(LSTMDataFormat.NST)
+                        .directionMode(LSTMDirectionMode.FWD)
+                        .gateAct(LSTMActivations.SIGMOID)
+                        .cellAct(LSTMActivations.TANH)
+                        .outAct(LSTMActivations.TANH)
+                        .retFullSequence(true)
+                        .retLastC(true)
+                        .retLastH(true)
+                        .build()
+        ).getAllOutputs();
```

Comment:
Does not compile - no getAllOutputs() method - let's use LSTMLayerOutputs instead of `List<SDVariable>`